src/www/ui/api/documentation/openapi.yaml

```diff
@@ -612,6 +612,96 @@ paths:
         default:
           $ref: '#/components/responses/defaultResponse'
 
+  /uploads/{id}/permissions:
+    parameters:
+      - name: id
+        required: true
+        description: Enter any upload id of that folder
+        in: path
+        schema:
+          type: integer
+    put:
+      operationId: setUploadPermissions
+      tags:
+        - Upload
+      summary: Set permissions for a upload in a folder for different groups
+      description: >
+        Set permissions for a upload in a folder for different groups
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                folderId:
+                  type: integer
+                  description: Enter a folder id
+                allUploadsPermission:
+                  type: string
+                  enum:
+                  - yes
+                  - no
+                  description: Apply same permissions to all uploads in this folder
+                newGroupId:
+                  type: integer
+                  description: Enter the id of the group you want to add or edit permission for this upload
+                newPermission:
+                  type: string
+                  enum:
+                  - none
+                  - read
+                  - write
+                  - clearing_admin
+                  - admin
+                  description: Select the permission for this group
+                publicPermission:
+                  type: string
+                  enum:
+                  - none
+                  - read
+                  - write
+                  - clearing_admin
+                  - admin
+                  description: Select the public permission for this upload
+              required:
+                - folderId
+                - allUploadsPermission
+      responses:
+        '200':
+          description: Permissions updated successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Info'
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Info'
+        '404':
+          description: Upload does not exist
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Info'
+        '409':
```

Comment:
Better response code will be [503 Service Unavailable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503)

Reply:
Okay!